src/main/groovy/com/codebrig/arthur/observe/structure/literal/RubyLiteral.groovy

```diff
@@ -9,21 +9,48 @@ import com.codebrig.arthur.observe.structure.StructureLiteral
  * @version 0.4
  * @since 0.2
  * @author <a href="mailto:brandon.fergerson@codebrig.com">Brandon Fergerson</a>
+ * @author <a href="mailto:valpecaoco@gmail.com">Val Pecaoco</a>
  */
 class RubyLiteral extends StructureLiteral {
 
     @Override
     String getNodeLiteralAttribute(SourceNode node) {
-        return null
+        switch (Objects.requireNonNull(node).internalType) {
+            case "int":
+                if (node.token.contains(".") || (node.token.isDouble() && node.token.toUpperCase().contains("E"))) {
+                    return doubleValueLiteral()
+                }
+                return numberValueLiteral()
+            case "float":
+                return doubleValueLiteral()
+            case "str":
+                return stringValueLiteral()
+            default:
+                return null
+        }
     }
 
     @Override
     List<String> getPossibleNodeLiteralAttributes(SourceNode node) {
-        def literalAttribute = getNodeLiteralAttribute(node)
-        if (literalAttribute == null) {
-            return Collections.emptyList()
-        } else {
-            return Collections.singletonList(literalAttribute)
+        switch (Objects.requireNonNull(node).internalType) {
+            case "int":
+                return [numberValueLiteral(), doubleValueLiteral()]
+            case "float":
+                return [doubleValueLiteral()]
+            case "str":
+                return [stringValueLiteral()]
+            default:
+                return Collections.emptyList()
         }
     }
+
+    static boolean isOctalLiteral(String literal) {
+        return literal.toUpperCase().contains("0O")
+    }
+
+    static Object getOctalValue(SourceNode node) {
+        boolean isNegative = node.parentSourceNode.children.any { it.roles.any { it.negative } }
```

Comment:
Should be using `isNodeLiteralNegative()` instead of inline

Reply:
Resolved with commit https://github.com/CodeBrig/Arthur/pull/79/commits/bcbacf6636d3ab2287458c86c10d09b357d97e1b